src/main/com/marginallyclever/makelangelo/paper/PaperSettings.java

```diff
@@ -51,13 +51,13 @@
 		new PaperSize("Arch E1",762,1067)
 	};
 
-	private Paper myPaper;
-	private SelectOneOfMany paperSizes;
-	private SelectDouble pw, ph, shiftX, shiftY, ang;
-	private SelectBoolean isLandscape;
-	private SelectSlider paperMargin;
-	private SelectColor paperColor;
-	
+	private final Paper myPaper;
+	private final SelectOneOfMany paperSizes;
+	private final SelectDouble pw, ph, shiftX, shiftY, ang;
+	private final SelectBoolean isLandscape;
+	private final SelectSlider paperMargin;
+	private final SelectColor paperColor;
```

Comment:
What does making these final do?

Reply:
`final` is used for variables only assigned in the constructor

Reply:
> Final Variables
> 
> When a variable is declared with final keyword, its value can’t be modified, essentially, a constant. This also means that you must initialize a final variable. If the final variable is a reference, this means that the variable cannot be re-bound to reference another object, but the internal state of the object pointed by that reference variable can be changed i.e. you can add or remove elements from the final array or final collection. It is good practice to represent final variables in all **uppercase**, using underscore to separate words.

  --  https://www.geeksforgeeks.org/final-keyword-in-java/

Reply:
yes, this a more detailed explanation. The final sentence `It is good practice to represent final variables in all uppercase, using underscore to separate words.` is not correct: variables whose are `static final` are constants, otherwise, they are just non-re-assignable value and may be different in each instance of the object. Thus, only `static final` should be uppercase.